src/components/Todo.js

```diff
@@ -0,0 +1,61 @@
+import { useContext, useState } from "react";
+import { TodosListContext } from "../contexts";
+import { faEdit, faSave, faTrash } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+export default function Todo({ title, id, deleteTodo }) {
+  const [isCheck, setIsCheck] = useState(false);
+  const [isEdit, setIsEdit] = useState(false);
+  const [editValue, setEditValue] = useState(title);
+  const { setData, data } = useContext(TodosListContext);
+  function editDataValue() {
+    let newData = data.map((item) => {
+      if (item.id === id) {
+        return {
+          ...item,
+          title: editValue,
+        };
+      }
+      return item;
+    });
+    setData(newData);
+    setIsEdit(!isEdit);
+  }
+  return (
+    <div className="flex justify-between items-center py-3 border-b">
+      {isEdit ? (
+        <div className="flex gap-10 items-center">
+          <input
+            className="border-2 border-black pt-1 pb-2 px-4 rounded-md"
+            onChange={(e) => setEditValue(e.target.value)}
+            value={editValue}
+          />
+          <FontAwesomeIcon
+            icon={faSave}
+            className="text-green-600 cursor-pointer"
+            onClick={editDataValue}
+          />
+        </div>
+      ) : (
+        <div className="flex gap-10 items-center">
+          <input type="checkbox" onClick={() => setIsCheck(!isCheck)} />
+          <p className={`${isCheck && "line-through"}`}>{title}</p>
+        </div>
+      )}
```

Comment:
You can use `contentEditable`, to make the content of any HTML tag editable, in this case you will not need to  have an input for editing and div for displaying the content.
link to the doc: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/contenteditable